src/c2pa/c2pa.py

```diff
@@ -2007,15 +2287,31 @@ def sign_file(self,
             raise C2paError.NotSupported(
                 f"Could not determine MIME type for file: {source_path}")
 
-        # Open source and destination files
-        with open(source_path, 'rb') as source_file, open(dest_path, 'wb') as dest_file:
-            # Convert Python streams to Stream objects
-            source_stream = Stream(source_file)
-            dest_stream = Stream(dest_file)
+        try:
+            # Open source file and create BytesIO for destination
+            with open(source_path, 'rb') as source_file:
+                source_stream = Stream(source_file)
 
-            # Use the internal stream-base signing logic
-            return self._sign_internal(
-                signer, mime_type, source_stream, dest_stream)
+                # In-memory buffer stream that will be flushed to target,
+                # to ensure proper in-memory processing and validatin
+                dest_stream = Stream(io.BytesIO(bytearray()))
```

Comment:
Why are we doing this in memory, the API accepts a stream. Is something broken or too slow?

Reply:
Something _was_ broken. Turns out target streams when opened need to be opened in w+b in Python (write, read and binary), not just write binary. Fixing that allowed me to do some refactoring too!